NODE_ENV not set when running shakapacker-dev-server

After upgrading to Shakapacker v9, I get the following error when trying to run `./bin/shakapacker-dev-server`:

```
[Shakapacker] Found Webpack config: [PROJECT_DIR]/config/webpack/webpack.config.js
[webpack-cli] Failed to load '[PROJECT_DIR]/config/webpack/webpack.config.js' config
[webpack-cli] Error: Cannot find module './undefined'
```

I'm still using an older shakapacker webpack configuration setup that defines custom webpack plugins in the development and production environments in custom `config/webpack/[environment].js` files, and this is my `webpack.config.js`:

```javascript
const webpackConfig = require(`./${process.env.NODE_ENV}`);

module.exports = webpackConfig;
```


## Code investigation:

In Shakapacker before v9, the `Shakapacker::DevServerRunner.run(ARGV)` command calls the parent Shakapacker::Runner's `run` class method which ensures the `ENV['NODE_ENV']` variable is set:
https://github.com/shakacode/shakapacker/blob/2ba54a34fee1bf818f72812daf7b8d71102ae264/lib/shakapacker/runner.rb#L6C1-L12C8


In Shakapacker v9, the ShakapackerDevServerRunner defines it's own [`run` class method](https://github.com/shakacode/shakapacker/blob/9fb968195d8b14b43b5dbe63e20af41b48584a3e/lib/shakapacker/dev_server_runner.rb#L11C1-L12C1) which does not appear to set a default `ENV['NODE_ENV']` environment variable, nor does it make a `Shakapacker.ensure_node_env!` call as is done in both the [Shakapacker::WebpackRunner](https://github.com/shakacode/shakapacker/blob/9fb968195d8b14b43b5dbe63e20af41b48584a3e/lib/shakapacker/webpack_runner.rb#L9) and the parent [Shakapacker::Runner](https://github.com/shakacode/shakapacker/blob/9fb968195d8b14b43b5dbe63e20af41b48584a3e/lib/shakapacker/runner.rb#L119) `run` class methods


I am able to resolve this by adding back in a `ENV['NODE_ENV'] ||= 'development'` line to my ./bin/shakapacker-dev-server script, but the documentation indicates this should not be necessary.

Is there a missing `Shakapacker.ensure_node_env!`invocation in the Shakapacker::DevServerRunner class, or should I be providing the NODE_ENV=development environment variable in some other way?

--- a/lib/shakapacker/dev_server_runner.rb
+++ b/lib/shakapacker/dev_server_runner.rb
@@ -18,6 +18,8 @@ module Shakapacker
         exit(0)
       end
 
+      Shakapacker.ensure_node_env!
+
       # Check for --build flag
       build_index = argv.index("--build")
       if build_index
@@ -65,6 +67,8 @@ module Shakapacker
     end
 
     def self.run_with_build_config(argv, build_config)
+      Shakapacker.ensure_node_env!
+
       # Apply build config environment variables
       build_config[:environment].each do |key, value|
         ENV[key] = value.to_s

--- a/spec/shakapacker/dev_server_runner_spec.rb
+++ b/spec/shakapacker/dev_server_runner_spec.rb
@@ -136,6 +136,37 @@ describe "DevServerRunner" do
     end
   end
 
+  describe "NODE_ENV environment variable" do
+    it "sets NODE_ENV when not already set" do
+      original_node_env = ENV.delete("NODE_ENV")
+
+      begin
+        Dir.chdir(test_app_path) do
+          klass = Shakapacker::DevServerRunner
+
+          allow(Shakapacker::Utils::Manager).to receive(:error_unless_package_manager_is_obvious!)
+
+          instance = klass.new([])
+
+          allow(klass).to receive(:new).and_return(instance)
+
+          # Stub build_cmd and system to prevent actual execution
+          allow(instance).to receive(:build_cmd).and_return(["webpack", "serve"])
+          allow(instance).to receive(:system) do |*args|
+            system("true")  # Sets $? to successful status
+            true
+          end
+
+          klass.run([])
+
+          expect(ENV["NODE_ENV"]).to eq("development")
+        end
+      ensure
+        ENV["NODE_ENV"] = original_node_env if original_node_env
+      end
+    end
+  end
+
   private
 
     def verify_command(cmd, argv: [], env: Shakapacker::Compiler.env)

